Add a section for order request culture providers (#24404)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/localization.md

Use `RequestLocalizationOptions` to add or remove localization providers.

### Change request culture providers order

> As mentioned earlier, `RequestLocalizationOptions` came up with three default request cultures provider: `QueryStringRequestCultureProvider`, `CookieRequestCultureProvider` and `AcceptLanguageHeaderRequestCultureProvider` respectively. In case you need to change the order of these providers you could use `RequestLocalizationOptions.RequestCultureProviders` property as shown in the example below:

Reviewer: ```suggestion
<xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptions> has three default request culture providers: <xref:Microsoft.AspNetCore.Localization.QueryStringRequestCultureProvider>, <xref:Microsoft.AspNetCore.Localization.CookieRequestCultureProvider>, and <xref:Microsoft.AspNetCore.Localization.AcceptLanguageHeaderRequestCultureProvider>. Use [`RequestLocalizationOptions.RequestCultureProviders`]](xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptions.RequestCultureProviders) property to change the order of these providers as shown in the following below:
```

```csharp
    app.UseRequestLocalization(options =>
    {
        var questStringCultureProvider = options.RequestCultureProviders[0];    
        options.RequestCultureProviders.RemoveAt(0);
        options.RequestCultureProviders.Insert(1, questStringCultureProvider);
    });
```

> In the example above, we switch between the `QueryStringRequestCultureProvider` and `CookieRequestCultureProvider`, so the `RequestLocalizationMiddleware` will look for the cultures from the cookies first, then query string. Perhaps the example is silly and simple, but image that you can order these providers with your custom once according your needs.

Reviewer: ```suggestion
In the preceding example above, the  order of `QueryStringRequestCultureProvider` and `CookieRequestCultureProvider` is swithched, so the `RequestLocalizationMiddleware` looks for the cultures from the cookies first, then query string.
```

> Another real use case - that we mentioned above - is when we add a custom provider via `AddInitialRequestCultureProvider` it sets the order to `0`, so this provider will take the precedence over the others.

Reviewer: ```suggestion
As previously mentioned, add a custom provider via <xref:Microsoft.AspNetCore.Builder.RequestLocalizationOptionsExtensions.AddInitialRequestCultureProvider%2A> which sets the order to `0`, so this provider takes the precedence over the others.
```
